uuid-cluster-dns/README.md

```diff
@@ -0,0 +1,79 @@
+# Using UUIDs for cluster endpoint domains
+
+> Assumption: this is relevant only going forward with CAPI clusters and not to be "backported" to the vintage product.
+
+## Current Approach
+
+The current approach for building cluster domains is to include the workload cluster's ID as a subdomain of the management cluster it belongs to, e.g.:
+
+```
+Workload Cluster: https://api.j9yr2.k8s.gauss.eu-west-1.aws.gigantic.io
+Management Cluster: https://dex.g8s.gauss.eu-west-1.aws.gigantic.io
+```
+
+The domain contains several parts:
+
+1. The name of the service being accessed (`api`)
+2. The workload cluster's ID, which can be user-provided (`j9yr2`) [Workload Cluster only]
+3. The static string `k8s` or `g8s`
+4. The codename of the management cluster (`gauss`)
+5. The region the cluster is deployed into (`eu-west-1`)
+6. The cloud provider (`aws`)
+7. The static string `gigantic.io`
+
+This approach has a few drawbacks:
+
+1. The workload cluster is tied to the management cluster (belonging to it via domain heirarchy) which is problamatic if the cluster needs migrating to another management cluster.
+2. Similar to above, workload clusters cannot be "promoted" to management clusters as the domain would no longer make sense.
+3. Several customers have their own domain set up, leading to inconsistency between MCs (e.g. `eu-central-1.aws.cps.customer.com` vs. `gauss.eu-west-1.aws.gigantic.io`).
```

Comment:
From @teemow 's comment I understood that we should rather not strive for supporting custom domains in the future.

Reply:
Updated to state that custom domains are a non-goal. 👍